etna/collections/models.py

```diff
@@ -205,6 +217,16 @@ def results_pages(self):
     ]
     subpage_types = ["collections.TimePeriodExplorerPage", "collections.ResultsPage"]
 
+    def get_context(self, request):
```

Comment:
Can you think of an alternative to overriding `get_context()`, that might be neater?

Reply:
This is what Gasman says : 
https://stackoverflow.com/questions/32626815/wagtail-views-extra-context

Reply:
Did you know you can access all fields values, methods and properties from the page directly with something like `{{ page.attribute_name }}`? If you broke the queryset stuff out into a separate method like this:

```
    @cached_property
    def related_insights(self):
        return InsightsPage.objects.filter(time_period=self).live().order_by("title")[:3]
```

You could easily access this from the template like so:

```
{% for obj in page.related_insights %}
    ...
{% endfor %}
```

Reply:
^ This also allows you to access the queryset for other purposes too (i.e. not just when rendering the page).

Reply:
Yes, it makes sense :)